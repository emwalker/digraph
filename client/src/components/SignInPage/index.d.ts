import React, { Component } from 'react';
type Props = {};
declare class SignInPage extends Component<Props> {
    renderSignInButton: () => React.JSX.Element;
    render: () => React.JSX.Element;
}
export default SignInPage;
