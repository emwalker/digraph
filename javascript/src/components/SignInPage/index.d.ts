import { Component } from 'react';
declare type Props = {};
declare class SignInPage extends Component<Props> {
    renderSignInButton: () => JSX.Element;
    render: () => JSX.Element;
}
export default SignInPage;
