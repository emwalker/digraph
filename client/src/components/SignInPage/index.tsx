import React, { Component } from 'react'

import Page from 'components/ui/Page'
import GithubLogin from 'components/ui/GithubLogin'

type Props = {}

class SignInPage extends Component<Props> {
  renderSignInButton = () => (
    <button
      className="btn btn-primary"
      disabled={false}
      type="button"
    >
      Sign in
    </button>
  )

  render = () => (
    <Page>
      <div className="SignInPage clearfix">
        <div className="col-10 col-md-6 column">
          <h2 className="mb-2">Sign in</h2>
          <p className="mb-2">Log in with your GitHub account:</p>
          <GithubLogin className="mb-5" />
        </div>
      </div>
    </Page>
  )
}

export default SignInPage
