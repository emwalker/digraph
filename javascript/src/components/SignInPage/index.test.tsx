import React from 'react'
import { shallow } from 'enzyme'

import SignInPage from './index'

const props = {}

describe('<SignInPage />', () => {
  const wrapper = shallow(<SignInPage {...props} />)

  it('renders', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
