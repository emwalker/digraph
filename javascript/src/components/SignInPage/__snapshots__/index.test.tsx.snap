// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignInPage /> renders 1`] = `
<Page>
  <div
    className="SignInPage clearfix"
  >
    <div
      className="col-10 col-md-6 column"
    >
      <h2
        className="mb-2"
      >
        Sign in
      </h2>
      <p
        className="mb-2"
      >
        Log in with your GitHub account:
      </p>
      <GithubLogin
        className="mb-5"
      />
    </div>
  </div>
</Page>
`;
